langworld_db_data/removers/listed_value_remover.py

```diff
@@ -30,55 +36,104 @@ def _remove_from_inventory_of_listed_values(
         id_of_value_to_remove: str,
     ) -> dict[str, str]:
         removed_value_information = {}
+        """
+        Contains the feature ID and English and Russian names of the removed value.
+        """
 
         rows = read_dicts_from_csv(self.input_file_with_listed_values)
 
         line_number_of_value_to_remove = 0
 
         for i, row in enumerate(rows):
-            if row["id"] == id_of_value_to_remove:
-                line_number_of_value_to_remove = i
-                removed_value_information["feature_id"] = row["feature_id"]
-                removed_value_information["en"] = row["en"]
-                removed_value_information["ru"] = row["ru"]
-                break
-
-        rows_with_updated_value_indices = self._update_value_indices_in_inventory(
-            rows=rows,
-            id_of_value_to_remove=id_of_value_to_remove,
-        )
+            if row["id"] != id_of_value_to_remove:
+                continue
+            line_number_of_value_to_remove = i
+            removed_value_information = {key: row[key] for key in ("feature_id", "en", "ru")}
 
         # If value not found, inventory will remain intact
         rows_without_removed_value = (
-            rows_with_updated_value_indices[:line_number_of_value_to_remove]
-            + rows_with_updated_value_indices[line_number_of_value_to_remove + 1 :]
+            rows[:line_number_of_value_to_remove] + rows[line_number_of_value_to_remove + 1 :]
+        )
+
+        rows_without_removed_value_and_with_updated_value_indices = (
+            self._update_value_indices_in_inventory(
+                rows=rows_without_removed_value,
+                id_of_removed_value=id_of_value_to_remove,
+            )
         )
 
         write_csv(
-            rows_without_removed_value,
+            rows_without_removed_value_and_with_updated_value_indices,
             path_to_file=self.output_file_with_listed_values,
             overwrite=True,
             delimiter=",",
         )
 
-        return removed_value_information
+        if not removed_value_information:
+            raise ListedValueRemoverError(
+                f"Value ID {id_of_value_to_remove} not found. "
+                f"Perhaps it is invalid or does not exist"
+            )
+        else:
+            return removed_value_information
+
+    def _remove_from_feature_profiles(
+        self,
+        id_of_value_to_remove: str,
+        english_name_of_value_to_remove: str,
```

Comment:
note that `english_name_of_value_to_remove` is not used in the code of this method, so you can remove it here and above (where you call this method).

Reply:
UPD I did it myself in e483bef